language_service/src/hover.rs

```diff
@@ -188,18 +189,35 @@ impl Visitor<'_> for HoverVisitor<'_> {
 }
 
 fn markdown_with_doc(doc: &Rc<str>, code: impl Display) -> String {
-    if doc.is_empty() {
+    let parsed_doc = parse_doc(doc);
+    if parsed_doc.is_empty() {
         markdown_fenced_block(code)
     } else {
         format!(
             "{}
 {}",
-            doc,
+            parsed_doc,
             markdown_fenced_block(code)
         )
     }
 }
 
+fn parse_doc(doc: &str) -> &str {
+    let summary_re =
```

Comment:
This seems like a rough way of accomplishing the goal. I'm open to suggestions on how to find the summary better.

Reply:
@sezna for suggestions

Reply:
One general suggestion from my side would be that the `parse_doc` function should return a `Documentation` struct rather than a string. This struct could be something like the following:
```rust
struct Documentation {
    summary: String,
    description: String,
    type_parameters: Vec<TypeParameter>,
    input: Vec<Input>,
    output: String,
    remarks: String,
    examples: Vec<Example>
}
```

Not everything has to be implemented in this PR (summary and description could be sufficient for now) but having this kind of structure in place would be useful for other hover/auto-complete features later on.

Returning a struct also has the benefit of providing context about what the `parse_doc` function does without having to read/understand the function. If it just returns a string, the person reading the code would have to read the function to know what the returned string represents. If it returns a struct, it's clear that the function creates structured data based on the input string without having to read or understand the function.

Reply:
I've implemented this struct. However, I am a bit concerned around over-indexing on establishing an API around how doc comments are written. I know the parameters should have a hook in the format, so that the hover can eventually use that when hovering over parameters, but other than that, I'd like to not have unnecessary structural expectations around doc comments. What are your thoughts on this?

Reply:
I think @sezna's suggestion is good. We can start with just having `summary` because it's the only one it's needed right now. As more stuff is needed, more fields could be added.

Reply:
simplified the Documentation structure to just the `summary` field.